autonat/README.md

```diff
@@ -0,0 +1,111 @@
+# NAT Discovery
+> How we dected if we're behind a NAT.
+
+| Lifecycle Stage | Maturity      | Status | Latest Revision |
+|-----------------|---------------|--------|-----------------|
+| 1A              | Recommendation | Active | r0, 2021-08-26  |
+
+
+Authors: [@marten-seemann]
+
+Interest Group: [@mxinden], [@vyzo], [@raulk], [@stebalien], [@willscott]
+
+[@marten-seemann]: https://github.com/marten-seemann
+[@mxinden]: https://github.com/mxinden/
+[@vyzo]: https://github.com/vyzo
+[@raulk]: https://github.com/raulk
+[@stebalien]: https://github.com/stebalien
+[@willscott]: https://github.com/willscott
+
+See the [lifecycle document][lifecycle-spec] for context about maturity level
+and spec status.
+
+[lifecycle-spec]: https://github.com/libp2p/specs/blob/master/00-framework-01-spec-lifecycle.md
+
+## Table of Contents
+
+TODO: fill in
+
+## Overview
+
+A priori, a node cannot know if it is behind a NAT or if it is publicly reachable.
```

Comment:
```suggestion
A priori, a node cannot know if it is behind a NAT / firewall or if it is publicly reachable.
```

AutoNAT is also relevant for nodes that are behind firewalls but not NATs. Should we stress that fact in this specification?


[Especially relevant, as **soon** (:innocent: ) the whole internet speaks IPv6, hopefully with little to no (IPv4toIPv6) NATs.](https://www.google.com/intl/en/ipv6/statistics.html)